app/src/main/java/learning/android/tenmarks/com/androidlearning/SettingsActivity.java

```diff
@@ -16,10 +18,22 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_settings);
 
+        Bundle extras = getIntent().getExtras();
+        String titleFromDashBoard;
+
+        if (extras != null) {
+            titleFromDashBoard = extras.getString("titleSentFromDashBoard");
+            TextView textView = (TextView) findViewById(R.id.text_view_center);
+            textView.setText(titleFromDashBoard);
+        }
+
         Button closeButton = (Button) findViewById(R.id.button_close);
         closeButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                Intent dashboardIntent = new Intent();
+                dashboardIntent.putExtra("backToTheDashBoard", "I'm back!");
+                setResult(123, dashboardIntent);
```

Comment:
a correction here: the first argument to setResult is either Activity.RESULT_OK or Activity.RESULT_CANCELED which is called ResultCode. The request code is automatically received to the caller activity when this callee returns data back to caller. 
Result Code gives your extra hold to handle the yes/no states in on result.